helper/threads.c

```diff
@@ -390,10 +431,10 @@ int odph_odpthreads_create(odph_odpthread_t *thread_tbl,
 		start_args->instance   = thr_params->instance;
 
 		if (helper_options.mem_model == ODP_MEM_MODEL_THREAD) {
-			if (create_pthread(&thread_tbl[i], cpu))
+			if (create_pthread(&thread_tbl[i], cpu, start_args->thr_params.stack_size))
 				break;
 		 } else {
-			if (create_process(&thread_tbl[i], cpu))
+			if (create_process(&thread_tbl[i], cpu, start_args->thr_params.stack_size))
```

Comment:
This is a backward incompatible change since the caller may not have memzeroed the thread parameter struct but just explicitly initialized the fields that existed in a previous version, leaving stack_size to an uninitialized value.

Since this function is deprecated, maybe its behaviour should not be changed in a backward incompatible manner. Then old code calling the function would not be at the risk of breaking. OTOH, if the calling code could be changed (e.g. to use the new parameter), then I suppose it should just starting using odph_thread_create() instead.

I think it would be good to stop using this function in ODP source tree, especially in the examples.

Otherwise the patches look good to me.

Reply:
> This is a backward incompatible change since the caller may not have memzeroed the thread parameter struct but just explicitly initialized the fields that existed in a previous version, leaving stack_size to an uninitialized value.

Yes, and the same applies to the other added parameter in this series as well.

> Since this function is deprecated, maybe its behaviour should not be changed in a backward incompatible manner.

Agree. But in that case, either the documentation for this function should state that stack_size is ignored, or stack_size should be added only to odph_thread_param_t, not odph_odpthread_params_t. The latter is currently just a typedef on the former. I'll come up with something.

> I think it would be good to stop using this function in ODP source tree, especially in the examples.

There are quite a few occurrences, but I can do that too.

Reply:
Fixed these in v2.